langchain/src/chains/combine_docs_chain.ts

```diff
@@ -439,6 +443,10 @@ export class RefineDocumentsChain
         runManager?.getChild("refine")
       );
       refineSteps.push(res);
+      if (this.onDocumentProcessed) {
+        const progress = (i / currentDocs.length).toFixed(2); // Progress calculation
+        await this.onDocumentProcessed(Number(progress)); // Emitting the progress
```

Comment:
Ah, was also suggesting that this emits the actual document in addition to just a fraction. User could calculate a `progress` meter on their own.

Let's actually hold off on this though - we're starting to look more at these older chains to see how we can modernize them more with LCEL principles.

Reply:
Right, instead of calculating and returning the progress, would make more sense to return the document itself,
and maybe the current loop number, It'd allow more freedom in the code side.

I understand that this is currently on hold to adapt and implement in LECL way, any tentative plan for when is this feature might be picked up? (I kind of need this for my project)

Thanks

Reply:
So sorry for missing this reply - could this work with the existing callback system by adding a `handleLLMEnd` callback to the model?

https://js.langchain.com/docs/modules/callbacks/how_to/create_handler